xcode/Subconscious/Shared/Components/Common/Profile/FollowUserFormSheet.swift

```diff
@@ -183,6 +194,42 @@ struct FollowNewUserFormSheetModel: ModelProtocol {
             var model = state
             model.failQRCodeScanErrorMessage = error
             return Update(state: model)
+            
+        case .failFollowDueToPetnameCollision(let error, let petname):
+            return update(
+                state: state,
+                actions: [
+                    .attemptToFindUniquePetname(petname: petname),
+                    .form(.failFollowDueToPetnameCollision(error))
+                ],
+                environment: environment
+            )
+            
+        case .attemptToFindUniquePetname(let petname):
+            let fx: Fx<FollowNewUserFormSheetAction> = environment
+                .addressBook
+                .findAvailablePetnamePublisher(name: petname)
+                .map { petname in
+                    .succeedFindUniquePetname(petname: petname)
+                }
+                .recover { error in
+                    .failToFindUniquePetname(error.localizedDescription)
+                }
+                .eraseToAnyPublisher()
+            
+            return Update(state: state, fx: fx)
+        case .succeedFindUniquePetname(let petname):
+            return update(
+                state: state,
+                action: .form(
+                    .petnameField(.setValue(input: petname.verbatim))
+                ),
+                environment: environment
+            )
+        case .failToFindUniquePetname(let error):
+            // Not a huge deal, the user will have to enter a name themselves
+            logger.warning("Failed to find a unique petname: \(error)")
+            return Update(state: state)
```

Comment:
Do we display anything to the user in this case?

Reply:
I'm guessing the answer is yes, via the `failFollowMessage`

Reply:
I would argue no based on my comment:

> Not a huge deal, the user will have to enter a name themselves

To be clearer: when this occurs the input field will already be marked as invalid and have the caption changed to the message "Petname already in use" (which is the `localizedDescription` for the relevant `AddressBookServiceError`.

Reply:
I'll update my in-code comment with this context.